bitvm/src/hash/blake3_utils.rs

```diff
@@ -495,7 +495,7 @@ pub(crate) fn compress(
         let mut tmp = Vec::new();
 
         //iterate nibbles
-        for n in 0..8 {
+        for n in 0..final_rounds {
```

Comment:
I'm not very familiar with this code, but wouldn't this push wrong values for `final_rounds > 8`? Since those values are supposed to be calculated differently? From the spec: 

<img width="1131" height="537" alt="image" src="https://github.com/user-attachments/assets/fe29e3f5-8092-4a5c-beb8-e0cf62d562de" />

Reply:
Agree. and the below expression `*state.get(&(i + 8)).unwrap();` would be panic due to accessing not-existed key.

I think we can add an assertion that final_rounds equals 8.